src/include/duckdb/common/types/string_type.hpp

```diff
@@ -117,10 +120,79 @@ struct string_t {
 
 	void Verify() const;
 	void VerifyNull() const;
+
+	struct StringComparisonOperators {
+		static inline bool Equals(const string_t &a, const string_t &b) {
+#ifdef DUCKDB_DEBUG_NO_INLINE
+			if (a.GetSize() != b.GetSize())
+				return false;
+			return (memcmp(a.GetDataUnsafe(), b.GetDataUnsafe(), a.GetSize()) == 0);
+#endif
+			uint64_t A;
+			uint64_t B;
+			memcpy(&A, &a, 8u);
+			memcpy(&B, &b, 8u);
+			if (A != B) {
+				// Either lenght or prefix are different -> not equal
+				return false;
+			}
+			// they have the same length and same prefix!
+			memcpy(&A, ((const char *)&a + 8u), 8u);
+			memcpy(&B, ((const char *)&b + 8u), 8u);
+			if (A == B) {
+				// either they are both inlined (so compare equal) or point to the same string (so compare equal)
+				return true;
+			}
+			if (!a.IsInlined()) {
+				// 'long' strings of the same length -> compare pointed value
+				if (memcmp(a.value.pointer.ptr, b.value.pointer.ptr, a.GetSize()) == 0) {
+					return true;
+				}
+			}
+			// either they are short string of same lenght but different content
+			//     or they point to string with different content
+			//     either way, they can't represent the same underlying string
+			return false;
+		}
+		// compare up to shared length. if still the same, compare lengths
+		static bool GreaterThan(const string_t &left, const string_t &right) {
+			const uint32_t left_length = left.GetSize();
+			const uint32_t right_length = right.GetSize();
+			const uint32_t min_lenght = std::min<uint32_t>(left_length, right_length);
+
+			if (min_lenght == 0u)
+				return (left_length > 0u);
+
+#ifdef DUCKDB_DEBUG_NO_INLINE
+			uint32_t A;
+			uint32_t B;
+			memcpy(&A, left.value.pointer.prefix, 4u);
+			memcpy(&B, left.value.pointer.prefix, 4u);
+
+			// Check on prefix -----
+			// We dont' need to mask since:
+			//	if the prefix is greater, it will stay greater regardless of the extra bytes
+			// 	if the prefix is smaller, it will stay smaller regardless of the extra bytes
+			//	if the prefix is equal, the extra bytes are guaranteed to be /0 for the shorter one
+
+			if (A != B)
+				return A > B;
+#endif
+			auto memcmp_res = memcmp(left.GetDataUnsafe(), right.GetDataUnsafe(), min_lenght);
+			auto final_res = (memcmp_res == 0) ? (left_length > right_length) : (memcmp_res > 0);
```

Comment:
Perhaps we can use predication instead of having a branch here, e.g.:
`memcmp_res > 0 || (memcmp_res == 0 && left_length > right_length)`